pkgs/development/libraries/mesa/default.nix

```diff
@@ -392,4 +394,6 @@ in stdenv.mkDerivation {
       substitutions.drivers = mesa.drivers;
     } ./llvmpipe-hook.sh;
   };
-}
+} // lib.optionalAttrs (stdenv.cc.bintools.isLLVM && lib.versionAtLeast stdenv.cc.bintools.version "17") {
+  NIX_LDFLAGS = "--undefined-version";
```

Comment:
Why?

Reply:
```
FAILED: src/gallium/targets/pipe-loader/pipe_i915.so 
x86_64-unknown-linux-gnu-clang++  -o src/gallium/targets/pipe-loader/pipe_i915.so src/gallium/targets/pipe-loader/pipe_i915.so.p/pipe_i915.c.o -Wl,--as-needed -Wl,--no-undefined -shared -fPIC -Wl,-soname,pipe_i915.so -Wl,--start-group src/gallium/auxiliary/libgallium.a src/compiler/nir/libnir.a src/compiler/libcompiler.a src/util/libmesa_util.a src/util/libmesa_util_sse41.a src/util/blake3/libblake3.a src/c11/impl/libmesa_util_c11.a src/gallium/auxiliary/libgalliumvl.a src/gallium/auxiliary/libgalliumvlwinsys.a src/gallium/drivers/i915/libi915.a src/gallium/winsys/i915/drm/libi915drm.a src/intel/common/libintel_common.a src/intel/isl/libisl.a src/intel/isl/libisl_per_hw_ver40.a src/intel/dev/libintel_dev.a src/intel/isl/libisl_per_hw_ver50.a src/intel/isl/libisl_per_hw_ver60.a src/intel/isl/libisl_per_hw_ver70.a src/intel/isl/libisl_per_hw_ver75.a src/intel/isl/libisl_per_hw_ver80.a src/intel/isl/libisl_per_hw_ver90.a src/intel/isl/libisl_per_hw_ver110.a src/intel/isl/libisl_per_hw_ver120.a src/intel/isl/libisl_per_hw_ver125.a src/intel/isl/libisl_per_hw_ver200.a src/intel/isl/libisl_tiled_memcpy.a src/intel/isl/libisl_tiled_memcpy_sse41.a src/intel/common/libintel_clflushopt.a -Wl,--gc-sections -Wl,--build-id=sha1 -Wl,--version-script /build/mesa-24.1.2/build/src/gallium/targets/pipe-loader/pipe_i915.sym /nix/store/s7f31asyzskvi3xf1162kp8b8jr37iz1-zlib-x86_64-unknown-linux-gnu-1.3.1/lib/libz.so -pthread -lm /nix/store/sfz7b94dm2s8ij49hiwc1spiv577x43j-zstd-x86_64-unknown-linux-gnu-1.5.6/lib/libzstd.so /nix/store/gkbfv526nmvqijllacngrh8a80k0lhy2-libunwind-x86_64-unknown-linux-gnu-1.8.1/lib/libunwind.so /nix/store/2bymvkqsxrjkbz0ydx0w9hn780qy4hwv-libdrm-x86_64-unknown-linux-gnu-2.4.121/lib/libdrm.so -L/nix/store/c5b927mai3l88qm5wq8nzfi5m9dnhbb8-llvm-x86_64-unknown-linux-gnu-17.0.6-lib/lib -lLLVM-17 -lsensors /nix/store/ajai5zrqpmdw6sc90rxizlhqjyamzrid-libxcb-x86_64-unknown-linux-gnu-1.17.0/lib/libxcb-sync.so /nix/store/ajai5zrqpmdw6sc90rxizlhqjyamzrid-libxcb-x86_64-unknown-linux-gnu-1.17.0/lib/libxcb-present.so /nix/store/shls4kmkps1yfgcdn543qk6xiqlgi645-libxshmfence-x86_64-unknown-linux-gnu-1.3.2/lib/libxshmfence.so /nix/store/ajai5zrqpmdw6sc90rxizlhqjyamzrid-libxcb-x86_64-unknown-linux-gnu-1.17.0/lib/libxcb-xfixes.so /nix/store/ajai5zrqpmdw6sc90rxizlhqjyamzrid-libxcb-x86_64-unknown-linux-gnu-1.17.0/lib/libxcb-dri3.so /nix/store/2bymvkqsxrjkbz0ydx0w9hn780qy4hwv-libdrm-x86_64-unknown-linux-gnu-2.4.121/lib/libdrm_intel.so -Wl,--end-group
x86_64-unknown-linux-gnu-ld: error: version script assignment of 'global' to symbol 'driver_descriptor' failed: symbol not defined
clang++: error: linker command failed with exit code 1 (use -v to see invocation)
```

Reply:
That just seems like an upstream bug?

Reply:
https://gitlab.freedesktop.org/mesa/mesa/-/merge_requests/25551 possibly relevant?

Reply:
Oh I thought it wasn't an upstream bug since we've had this issue with multiple packages.

Reply:
It's basically always an upstream issue, it's trying to do things to symbols that don't exist. It's not fatal on ld.bfd by default, and lld only recently made it fatal, but it's always _wrong_.

Reply:
Yeah, though I know in one or maybe two cases where upstream refused to fix it.

Reply:
Alright, switched to using the patch

Reply:
Should probably apply the patch unconditionally, so that we can tell when it breaks.

Reply:
Oh ok, I was trying to prevent mass rebuilds. I'll switch it to be unconditional.